redaxo/src/core/tests/var/var_test.php

```diff
@@ -191,4 +191,30 @@ public function testQuote()
 
         $this->assertEquals($expected, rex_var_test_var::quote($string));
     }
+
+    public function testVarIterator()
+    {
+        $content = '
+            hallo
+            REX_TEST_VAR[content="ab" suffix=ef]
+            welt
+            REX_TEST_VAR[content="cd" suffix="gh"]
+            123
+        ';
+
+        $it = rex_var::varsIterator($content);
+        $firstVar = $it->current();
+        $this->assertInstanceOf(rex_var_test_var::class, $firstVar);
+        $this->assertSame("'ab'", $firstVar->getOutput());
+        $this->assertSame("ef", $firstVar->getArg('suffix'));
```

Comment:
oder sollten wir stattdessen `getParsedArg()` verwenden?

Reply:
Eigentlich bestimmt die jeweilige Rex-Vars, welche Args wie eingelesen werden.
`getArg` liefert das Argument als direkt verwertbaren Wert (während der Erstellung des Caches), unterstützt aber zum Beispiel keine nested Rex-Vars.
`getParsedArg` hingegen unterstützt nested Rex-Vars, liefert dann aber eine PHP-Expression, die man in der eigenen PHP-Ausgabe einbetten kann. Der tatsächliche Wert liegt somit während der Erstellung der Cache-Datei noch nicht zur Verfügung (da man zu dem Zeitpunkt eben nur einen PHP-Code-Schnipsel erhält), sondern erst beim  Ausführen der Cache-Datei.

Reply:
Oder anders gesagt: Die Rex-vars sollten möglichst mit `getParsedArg` arbeiten, weil dann eben auch nested Vars möglich sind. Und nur, wenn es unbedingt sein muss, und bereits während der Erstellung des Caches der konkrete Wert ausgewertet werden muss, sollte `getArg` verwendet werden.